protocol/signalfx/datapoint.go

```diff
@@ -262,3 +264,70 @@ func SetupJSONV1Paths(r *mux.Router, handler http.Handler) {
 	SetupJSONByPaths(r, handler, "/datapoint")
 	SetupJSONByPaths(r, handler, "/v1/datapoint")
 }
+
+// ErrorTrackerHandler behaves like a http handler, but tracks error returns from a ErrorReader
+type ErrorTrackerHandler struct {
+	TotalErrors int64
+	reader      ErrorReader
+	Logger      log.Logger
+}
+
+// Datapoints gets TotalErrors stats
+func (e *ErrorTrackerHandler) Datapoints() []*datapoint.Datapoint {
+	return []*datapoint.Datapoint{
+		sfxclient.Cumulative("total_errors", nil, atomic.LoadInt64(&e.TotalErrors)),
+	}
+}
+
+func addTokenToContext(ctx context.Context, req *http.Request) context.Context {
+	head := req.Header.Get(sfxclient.TokenHeaderName)
+	if head != "" {
+		ctx = context.WithValue(ctx, sfxclient.TokenHeaderName, head)
+	}
+	return ctx
+}
+
+// ServeHTTPC will serve the wrapped ErrorReader and return the error (if any) to rw if ErrorReader
```

Comment:
What does the `C` stand for in `ServeHTTPC`?

Reply:
I don't know tbh. Just restored the recently removed code that is apparently still being used